projects/plugins/social/src/class-note.php

```diff
@@ -46,8 +76,34 @@ public static function register() {
 			'show_in_rest' => true,
 			'supports'     => array( 'editor', 'thumbnail', 'publicize', 'activitypub' ),
 			'menu_icon'    => 'dashicons-welcome-write-blog',
-
+			'rewrite'      => array( 'slug' => 'sn' ),
 		);
-		register_post_type( 'jetpack-social-note', $args );
+		register_post_type( self::JETPACK_SOCIAL_NOTE_CPT, $args );
+		static::maybe_flush_rewrite_rules();
+	}
+
+	/**
+	 * Flush rewrite rules so the post permalink works correctly for the Social Note CPT. Flushing is an expensive operation, so do only when necessary.
+	 *
+	 * @param boolean $force Force flush the rewrite rules.
+	 */
+	public static function maybe_flush_rewrite_rules( $force = false ) {
+		if ( empty( get_option( self::FLUSH_REWRITE_RULES_FLUSHED ) ) || $force ) {
+			flush_rewrite_rules( false );
+			update_option( self::FLUSH_REWRITE_RULES_FLUSHED, true );
+		}
+	}
+
+	/**
+	 * Toggle whether or not the Notes feature is enabled.
+	 */
+	public static function toggle_enabled_status() {
```

Comment:
This function is only used in tests now, I thought the UI toggle could be a task of its own.

Reply:
Yes definitely! Having the function ready to link up to the UI is as much as we want to do here